notable/SigninViewController.swift

```diff
@@ -100,7 +51,94 @@ class SigninViewController: UIViewController, UIScrollViewDelegate {
         
         // definesPresentationContext = true
         
-        // Do any additional setup after loading the view.
+    }
+    
+    @IBAction func didTouchBackButton(sender: AnyObject) {
+         performSegueWithIdentifier("signinIntroSegue", sender:self)
+    }
+
+    @IBAction func didTouchNeedSignupButton(sender: AnyObject) {
+        performSegueWithIdentifier("needSignUpSegue", sender:self)
+        
+    }
+    
+    var initialYtitleLabel: CGFloat!
+    var offsettitleLabel: CGFloat!
+    
+    var initialYfieldParentView: CGFloat!
+    var offsetfieldParentView: CGFloat!
+    
+    var initialYbuttonParentView: CGFloat!
+    var offsetbuttonParentView: CGFloat!
+    
+    var defaults = NSUserDefaults.standardUserDefaults()
+    
+    func showError(error: String) {
+        // errorMessage.hidden = false
+        // errorMessage.text = error
+        
+    }
+    
+    @IBAction func onEmailChange(sender: AnyObject) {
+        //        validateForm()
+    }
+    
+    @IBAction func onPasswordChange(sender: AnyObject) {
+        //        validateForm()
+    }
+    
+    //    func validateForm(){
+    //        if (passwordInput.text != "" && emailInput.text != "") {
+    //            loginButton.enabled = true
+    //            print("loginButton enabled")
+    //        }
+    //    }
+    //
+    
+    @IBAction func didPressSignin(sender: AnyObject) {
+  
+//        PFUser.logInWithUsernameInBackground(emailInput.text!, password: passwordInput.text!) {
+//            (user: PFUser?, error: NSError?) -> Void in
+//            if user != nil {
+//                self.performSegueWithIdentifier("loginSegue", sender: self)
+//            } else {
+//                print(error)
+//            }
+//        }
+        
+        // if emailField.text == "asdf" && passwordField.text == "asdf"
+        
+        PFUser.logInWithUsernameInBackground(emailField.text!, password: passwordField.text!) {
```

Comment:
the PFUser block is what actually logs in, we shouldn't be doing the validation inside this block